mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -0,0 +1,247 @@
+
+        eventBus = new RabbitMQEventBus(connectionFactory, eventSerializer);
+        eventBus2 = new RabbitMQEventBus(connectionFactory, eventSerializer);
+        eventBus.start();
+        eventBus2.start();
+        sender = RabbitFlux.createSender(new SenderOptions().connectionMono(connectionMono));
+    }
+
+    @AfterEach
+    void tearDown() {
+        eventBus.stop();
+        eventBus2.stop();
+        ALL_GROUPS.stream()
+            .map(groupClass -> GroupRegistration.WorkQueueName.of(groupClass).asString())
+            .forEach(queueName -> sender.delete(QueueSpecification.queue(queueName)).block());
+        sender.delete(ExchangeSpecification.exchange(MAILBOX_EVENT_EXCHANGE_NAME)).block();
+        sender.close();
+    }
+
+    @Override
+    public EventBus eventBus() {
+        return eventBus;
+    }
+
+    @Override
+    public EventBus eventBus2() {
+        return eventBus2;
+    }
+
+    @Nested
+    class PublishingTest {
+        private static final String MAILBOX_WORK_QUEUE_NAME = MAILBOX_EVENT + "-workQueue";
+
+        @BeforeEach
+        void setUp() {
+            createQueue();
+        }
+
+        private void createQueue() {
+            SenderOptions senderOption = new SenderOptions()
+                .connectionMono(Mono.fromSupplier(connectionFactory::create));
+            Sender sender = RabbitFlux.createSender(senderOption);
+
+            sender.declareQueue(QueueSpecification.queue(MAILBOX_WORK_QUEUE_NAME)
+                .durable(DURABLE)
+                .exclusive(!EXCLUSIVE)
+                .autoDelete(!AUTO_DELETE)
+                .arguments(NO_ARGUMENTS))
+                .block();
+            sender.bind(BindingSpecification.binding()
+                .exchange(MAILBOX_EVENT_EXCHANGE_NAME)
+                .queue(MAILBOX_WORK_QUEUE_NAME)
+                .routingKey(EMPTY_ROUTING_KEY))
+                .block();
+        }
+
+        @Test
+        void dispatchShouldPublishSerializedEventToRabbitMQ() {
+            eventBus.dispatch(EVENT, NO_KEYS).block();
+
+            assertThat(dequeueEvent()).isEqualTo(EVENT);
+        }
+
+        @Test
+        void dispatchShouldPublishSerializedEventToRabbitMQWhenNotBlocking() {
+            eventBus.dispatch(EVENT, NO_KEYS);
+
+            assertThat(dequeueEvent()).isEqualTo(EVENT);
+        }
+
+        private Event dequeueEvent() {
+            RabbitMQConnectionFactory connectionFactory = rabbitMQExtension.getConnectionFactory();
+            Receiver receiver = RabbitFlux.createReceiver(new ReceiverOptions().connectionMono(Mono.just(connectionFactory.create())));
+
+            byte[] eventInBytes = receiver.consumeAutoAck(MAILBOX_WORK_QUEUE_NAME)
+                .blockFirst()
+                .getBody();
+
+            return eventSerializer.fromJson(new String(eventInBytes, StandardCharsets.UTF_8))
+                .get();
+        }
+    }
+
+    @Nested
+    class ConcurrentTest {
```

Comment:
Why are these concurrent tests not part of the Contract?

Reply:
Move to `EventBusConcurrentTestContract`